tools/tt-alchemist/csrc/python_runner/python_runner.hpp

```diff
@@ -0,0 +1,52 @@
+// SPDX-FileCopyrightText: (c) 2025 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef TT_ALCHEMIST_PYTHON_RUNNER_HPP
+#define TT_ALCHEMIST_PYTHON_RUNNER_HPP
+
+#include <string>
+#include <vector>
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wcovered-switch-default"
+#include "ttnn/device.hpp"
+#include "ttnn/tensor/tensor.hpp"
+#include <pybind11/embed.h>
+#include <pybind11/pybind11.h>
+#pragma clang diagnostic pop
+
+namespace tt::alchemist {
+
+/// Python model runner for executing generated Python code.
+///
+/// Usage:
+///   PythonModelRunner runner;
+///   runner.addToPath("/path/to/generated/model");
+///   runner.loadModule("main", "forward");
+///   auto outputs = runner.forward(inputs, device);
+///
+class PythonModelRunner {
+public:
+  PythonModelRunner();
+  ~PythonModelRunner() = default;
+
+  /// Add a directory to Python's sys.path for module imports.
+  void addToPath(const std::string &path);
```

Comment:
```suggestion
  void addToSysPath(const std::string &path);
```

This makes it a little more descriptive